pages/curriculum/[lessonSlug]/mentor/index.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useMemo, useState } from 'react'
+import {
+  AddExerciseMutation,
+  GetAppProps,
+  Module,
+  useAddExerciseMutation,
+  withGetApp
+} from '../../../../graphql'
+import { useRouter } from 'next/router'
+import { AdminLayout } from '../../../../components/admin/AdminLayout'
+import { DropdownMenu } from '../../../../components/DropdownMenu'
+import { FormCard, MD_INPUT } from '../../../../components/FormCard'
+import { formChange } from '../../../../helpers/formChange'
+import { exercisesValidation } from '../../../../helpers/formValidation'
+import ExercisePreview from '../../../../components/ExercisePreview'
+import styles from '../../../../scss/mentorPage.module.scss'
+import { get } from 'lodash'
+import { ApolloError } from '@apollo/client'
+import QueryInfo from '../../../../components/QueryInfo'
+import { errorCheckAllFields } from '../../../../helpers/admin/adminHelpers'
+import * as Sentry from '@sentry/nextjs'
+
+type DetachedModule = Omit<Module, 'lesson' | 'author'>
+
+type HeaderProps<T> = {
+  lesson?: T
+  addExerciseData?: AddExerciseMutation | null
+  loading: boolean
+  error?: ApolloError
+  setModule: (v: null | DetachedModule) => void
+}
+const Header = <
+  T extends { title: string; modules?: DetachedModule[] | null }
+>({
+  lesson,
+  addExerciseData,
+  loading,
+  error,
+  setModule
+}: HeaderProps<T>) => {
+  const modules = get(lesson, 'modules')
```

Comment:
can you try setting default value for modules as empty array `[]` incase `get(lesson, 'modules')` is falsey?

Reply:
I feel like TypeScript should have caught this error at compile time. Is `get(lesson, 'modules')` type-safe? If not, should we use `lesson?.modules` instead since it's type-safe?

Reply:
> I feel like TypeScript should have caught this error at compile time. Is `get(lesson, 'modules')` type-safe? If not, should we use `lesson?.modules` instead since it's type-safe?

Which error?

Reply:
> Which error?

Oh I made a mistake. There's no error because there's code below that checks if `modules` is truthy.

Reply:
Actually, a better question for me to ask is: is there a possibility that `lesson.modules` will be falsey? Or is it guaranteed(through typescript rules) to always exist in this `Header` component?

Reply:
> Actually, a better question for me to ask is: is there a possibility that `lesson.modules` will be falsey? Or is it guaranteed(through typescript rules) to always exist in this `Header` component?

Yes, there's a possibility for the `lesson.modules` to be falsey. The Header prop type is: `{ ..., modules?: DetachedModule[] | null }` which is the default type for the [lesson modules](https://github.com/flacial/c0d3-app/blob/f07fbca77acfd27c3bb5beb650ebab04fd799f83/graphql/index.tsx#L741-L747)